Bootstrap/Update.php

```diff
@@ -466,4 +467,47 @@ private function createRemoteToLocalCategoriesTable()
             }
         }
     }
+
+    private function recreateRemoteCategoriesAndProductAssignments()
+    {
+        if (version_compare($this->version, '1.1.4', '<=')) {
+            try {
+                $result = $this->db->query('SELECT `article_id`, `category` FROM `s_plugin_connect_items` WHERE shop_id IS NOT NULL');
+
+                while ($row = $result->fetch()) {
+                    $categories = json_decode($row['category'], true);
+                    $countAssignedCategories = $this->db->query('SELECT COUNT(`connect_category_id`) AS categories_count FROM s_plugin_connect_product_to_categories WHERE articleID = ?',
+                        [$row['article_id']]
+                    )->fetch();
+
+                    if (count($categories) != $countAssignedCategories['categoriesCount']) {
+                        foreach ($categories as $categoryKey => $category) {
+                            $result = $this->db->query('SELECT `id` FROM s_plugin_connect_categories WHERE category_key = ?',
+                                [$categoryKey]);
+                            if (!($res = $result->fetch())) {
+                                $this->db->query('INSERT INTO s_plugin_connect_categories (category_key, label) VALUES (?, ?)',
+                                    [$categoryKey, $category]);
+                                $categoryId = $this->db->lastInsertId('s_plugin_connect_categories');
+                            } else {
+                                $categoryId = $res['id'];
+                            }
+                            $result = $this->db->query('SELECT `id` FROM s_plugin_connect_product_to_categories WHERE connect_category_id = ? AND articleID = ?',
```

Comment:
$result.

Please do not duplicate names of the first statement which you use for while loop